src/python/pants/backend/python/framework/stevedore/rules.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from collections import defaultdict
+
+from pants.backend.python.framework.stevedore.target_types import (
+    ResolvedStevedoreEntryPoints,
+    ResolveStevedoreEntryPointsRequest,
+    StevedoreEntryPoints,
+    StevedoreEntryPointsField,
+    StevedoreNamespaceField,
+    StevedoreNamespacesField,
+)
+from pants.backend.python.goals.pytest_runner import PytestPluginSetup, PytestPluginSetupRequest
+from pants.backend.python.target_types import PythonTestsDependenciesField
+from pants.engine.fs import CreateDigest, Digest, FileContent, PathGlobs, Paths
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import DependenciesRequest, Target, Targets
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+
+class GenerateEntryPointsTxtFromStevedoreExtensionRequest(PytestPluginSetupRequest):
+    @classmethod
+    def is_applicable(cls, target: Target) -> bool:
+        # select python_tests targets with stevedore_namespaces field
+        return (
+            target.has_field(StevedoreNamespacesField)
+            and target.get(StevedoreNamespacesField).value is not None
```

Comment:
Note to Pants maintainers, not particularly for this PR.

As `target.get()` will return a default valued field if the target doesn't have one, checking for `target.has_field()` is then only an optimization to avoid creating a default instance, however at the cost of a double field lookup. Maybe worth it in this case to avoid thrashing the GC in case of many many targets but otherwise it would be interesting to run some benchmarks on the effectiveness of this.

My hunch is that we may want to make the `Target._maybe_get()` public for uses like this, where we don't want a default instance of a field in case it is not declared on the target. (or as a new `default=None` kwarg to `Target.get()`)